x/stakeibc/keeper/host_zone.go

```diff
@@ -112,6 +113,55 @@ func (k Keeper) GetAllHostZone(ctx sdk.Context) (list []types.HostZone) {
 	return
 }
 
+// Unregisters a host zone, including removing the module accounts and records
+func (k Keeper) UnregisterHostZone(ctx sdk.Context, chainId string) error {
+	hostZone, found := k.GetHostZone(ctx, chainId)
+	if !found {
+		return types.ErrHostZoneNotFound.Wrapf("host zone %s not found", chainId)
+	}
+
+	// Remove module accounts
+	depositAddress := types.NewHostZoneDepositAddress(chainId)
+	communityPoolStakeAddress := types.NewHostZoneModuleAddress(chainId, CommunityPoolStakeHoldingAddressKey)
+	communityPoolRedeemAddress := types.NewHostZoneModuleAddress(chainId, CommunityPoolRedeemHoldingAddressKey)
+
+	k.AccountKeeper.RemoveAccount(ctx, k.AccountKeeper.GetAccount(ctx, depositAddress))
+	k.AccountKeeper.RemoveAccount(ctx, k.AccountKeeper.GetAccount(ctx, communityPoolStakeAddress))
+	k.AccountKeeper.RemoveAccount(ctx, k.AccountKeeper.GetAccount(ctx, communityPoolRedeemAddress))
+
+	// Remove all deposit records for the host zone
+	for _, depositRecord := range k.RecordsKeeper.GetAllDepositRecord(ctx) {
```

Comment:
I understand that this function will often be called when we are actually keeping the data in a different way and doing an upgrade or something, but in case this remove is called on a hostzone without replacement, do we want to error or at least log if there are are records still because this means we have control of funds and we are losing the records which track them?

Reply:
I see that this is only expected to be called on the testnet, but how do we ensure that is the case?

Reply:
we ensure it's only called on testnet by not creating a tx for it